runtime/jvmti/jvmtiClass.c

```diff
@@ -1367,15 +1381,20 @@ copyInitiatedClass(J9Class * clazz, J9JVMTIClassStats * results)
 
 		J9JavaVM * vm = results->vm;
 		JNIEnv * jniEnv = (JNIEnv *)results->currentThread;
-		jint slot = (jint)results->classCount - 1;
-
-		/* Reverse fill */
-		if (slot >= 0) {
-			j9object_t classObject = J9VM_J9CLASS_TO_HEAPCLASS(clazz);
-			
-			results->classRefs[slot] = (jclass)vm->internalVMFunctions->j9jni_createLocalRef(jniEnv, classObject);
-			results->classCount = slot;
-		}
+
+		do {
+			jint slot = (jint)results->classCount - 1;
+
+			/* Reverse fill */
+			if (slot >= 0) {
+				j9object_t classObject = J9VM_J9CLASS_TO_HEAPCLASS(clazz);
+				
+				results->classRefs[slot] = (jclass)vm->internalVMFunctions->j9jni_createLocalRef(jniEnv, classObject);
```

Comment:
Can you cache the `vm->internalVMFunctions` table in a local?  A minor thing but might be a small benefit for deep array chains...

Reply:
Done.